app/models/document.rb

```diff
@@ -33,18 +19,38 @@ def path
   end
 
   def summary
-    nil
+    description = attrs.fetch(:description, nil)
+
+    # This truncates the description at the end of the first sentence
+    description = description.gsub(/\.\s[A-Z].*/, '.') if description.present? && finder.show_summaries?
+  end
+
+  def date_metadata_keys
+    finder.metadata.select{ |f| f.type == "date" }.map(&:key)
+  end
+
+  def tag_metadata_keys
+    finder.metadata.select{ |f| f.type == "text" }.map(&:key)
+  end
+
+  def metadata_keys
+    tag_metadata_keys + date_metadata_keys
+  end
+
+  def metadata_name_mappings
```

Comment:
Actually, we only want this to use facets which have `short_name`s, but I get what you're saying.